kinto/views/buckets.py

```diff
@@ -23,12 +26,20 @@ def get_parent_id(self, request):
         # Buckets are not isolated by user, unlike Cliquet resources.
         return ''
 
-    def delete(self):
-        result = super(Bucket, self).delete()
+
+@subscriber(ResourceChanged,
+            for_resources=('bucket',),
+            for_actions=(ACTIONS.DELETE,))
+def on_buckets_deleted(event):
+    """Some buckets were deleted, delete sub-resources.
+    """
+    storage = event.request.registry.storage
+
+    for change in event.impacted_records:
+        bucket = change['old']
+        parent_id = '/buckets/%s' % bucket['id']
```

Comment:
erf we should, but not only here...

http://makina-corpus.com/blog/metier/the-worlds-simplest-python-template-engine :)